compiler/x/codegen/OMRTreeEvaluator.cpp

```diff
@@ -3371,6 +3371,244 @@ static void arrayCopyDefault(TR::Node* node, uint8_t elementSize, TR::Register*
       }
    }
 
+/*
+ * Generate a store instruction
+ * Parameters:
+ * node: the tree node
+ * addressReg: the base register for the destination mem address
+ * index: the index for the destination mem address
+ * valueReg: the value that needs be stored
+ * size: the size of valueReg
+ * cg: the code generator
+ */
+static void generateStoreInstructions(TR::Node* node, TR::Register* addressReg, int32_t index, TR::Register* valueReg, uint8_t size,  TR::CodeGenerator* cg)
+   {
+   TR_X86OpCodes storeOpcode;
+   if (valueReg->getKind() == TR_FPR)
+      {
+      switch (size)
+         {
+         case 4:
+            storeOpcode = MOVDMemReg;
+            break;
+         case 8:
+            storeOpcode = MOVQMemReg;
+            break;
+         case 16:
+            storeOpcode = MOVDQUMemReg;
+            break;
+         }
+      }
+   else 
+      {
+      switch (size)
+         {
+         case 1:
+            storeOpcode = S1MemReg;
+            break;
+         case 2:
+            storeOpcode = S2MemReg;
+            break;
+         case 4:
+            storeOpcode = S4MemReg;
+            break;
+         case 8:
+            storeOpcode = S8MemReg;
+            break;
+
+         }
+      }
+   generateMemRegInstruction(storeOpcode, node, generateX86MemoryReference(addressReg, index, cg), valueReg, cg);
+   }
+
+/*
+ * Generate a load instruction
+ * Parameters:
+ * node: the tree node
+ * valueReg: the destination register
+ * size: the size of the loaded value
+ * addressReg: the base register for the source mem address
+ * index: the index for the source mem address
+ * cg: the code generator
+ */
+static void generateLoadInstructions(TR::Node* node, TR::Register* valueReg, uint8_t size, TR::Register* addressReg, int32_t index, TR::CodeGenerator* cg)
```

Comment:
There seems to be a very similar structure to the store case - any particular reason you chose not to combine them in some way? Is this more clear? Just seems like it could be a bit of duplication...